src/couch_views/src/couch_views_http.erl

```diff
@@ -177,18 +177,23 @@ maybe_add_next_bookmark(OriginalLimit, PageSize, Args0, Response, Items, KeyFun)
 
 maybe_add_previous_bookmark(#mrargs{extra = Extra} = Args, #{rows := Rows} = Result, KeyFun) ->
     StartKey = couch_util:get_value(fk, Extra),
-    case {StartKey, first_key(KeyFun, Rows)} of
-        {undefined, _} ->
+    StartId = couch_util:get_value(fid, Extra),
+    case {{StartId, StartKey}, first_key(KeyFun, Rows)} of
+        {{undefined, undefined}, _} ->
```

Comment:
Maybe use `{_, _}` instead of `_` for `first_key(...)` result matching just to assert that it always return a tuple of 2 elements